Add info about color and intensity to surface guide (#993)
napari/docs: docs/howtos/layers/surface.md

The surface layer allows you to display a precomputed surface mesh that is
defined by an `NxD` array of `N` vertices in `D` coordinates, an `Mx3` integer

> array of the indices of the triangles making up the faces of the surface, and,

Reviewer: Since there are several long components and one optional, changing to a bulleted list would reduce cognitive load for reader.

optionally, a length `N` list of values to associate with each vertex to use
alongside a colormap. If you omit the vertex values, napari uses a length `N`

> array of ones.

Reviewer 2: Actually, surfaces values can be n-dimensional:

https://github.com/napari/napari/blob/93ef1fbc8de2758fd1b381a5ef7e5508323547a4/src/napari/layers/surface/surface.py#L55-L58

Author: Can you please provide a suggestion :) Reading the docstring is not bringing clarity to us in the docs meeting 💔 We should probably update that docstring too

Author: RIP Me: 
<img width="463" height="130" alt="image" src="https://github.com/user-attachments/assets/7d8346fa-09c9-4e20-8d43-f3906f0f82f6" />

Can you please provide a suggestion :) Reading the docstring is not bringing clarity to us in the docs meeting 💔 We should probably update that docstring too

Reviewer 2: Ouch, sorry about this one! I'll come up with something now.

> array of ones.

Reviewer 2: ```suggestion
array of the indices of the triangles making up the faces of the surface, and,
optionally, a length `N` array of per-vertex values that can be used alongside a colormap to color the surface. Vertex values can also have additional leading dimensions (to represent, for example, a time series), in the form `(D1, D2, ..., N), which will be treated as extra dimensions by the viewer.
```

Once you have created a `surface` layer programmatically, the following GUI
controls are available in the viewer:

> Controls related to `contrast limits`, `auto-contrast`, `gamma`, and
> `colormap` are only available when the surface color is computed from
> `vertex_values`. If you provide `vertex_colors`, napari uses those colors

Reviewer: ```suggestion
When the surface color is computed from `vertex_values`, controls related
to `contrast limits`, `auto-contrast`, `gamma`, and
`colormap` are unavailable. If you provide `vertex_colors`, napari uses those colors
```

Is vertex_colors a vertex_value?

Author: so this is IMO a bigger problem than I can address in the PR. 
`vertex_values` is really the 3rd element of the data tuple, but it seems to be used throughout the docs and examples as if it were a real argument.
On the other hand, `vertex_colors` is an argument to `Surface` that overrides anything set in that third element of the data tuple. 
I'll take another overall swing at the wording -- I think bulleting the tuple elements like above will probably help with this :)

Reviewer: No problem. I just didn't know the answer. Keep it simple now with just the bullets.

## Surface data

The data for a `surface` layer can be given as either a 2-tuple
`(vertices, faces)` or a 3-tuple `(vertices, faces, vertex_values)`. The
vertices are an `NxD` array of `N` vertices in `D` coordinates. The faces are
an `Mx3` integer array of the indices of the triangles making up the faces of

> the surface. The optional vertex values are a length `N` list of scalar values

Reviewer 2: same here as above.

to associate with each vertex for colormap-based rendering. If you omit
`vertex_values`, napari fills them with ones, so the default underlying surface

> color is white.

Reviewer: Consider a bullet list for sentences after the first.

```suggestion
The data for a `surface` layer can be given as either a 2-tuple
`(vertices, faces)` or a 3-tuple `(vertices, faces, vertex_values)`.
- The vertices are an `NxD` array of `N` vertices in `D` coordinates. 
- The faces are an `Mx3` integer array of the indices of the triangles
  making up the faces of the surface. 
- The optional vertex values are a length `N` list of scalar values
  to associate with each vertex for colormap-based rendering. 
  If you omit `vertex_values`, napari fills them with ones, so the default
  underlying surface color is white.
```
